kolibri_explore_plugin/views.py

```diff
@@ -2,11 +2,54 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+import os
+import zipfile
+
+from django.http import Http404
+from django.http import HttpResponse
 from django.utils.decorators import method_decorator
+from django.views.decorators.clickjacking import xframe_options_exempt
 from django.views.generic.base import TemplateView
+from django.views.generic.base import View
+from kolibri.core.content.api import cache_forever
+from kolibri.core.content.decorators import add_security_headers
+from kolibri.core.content.views import get_embedded_file
 from kolibri.core.decorators import cache_no_user_data
 
 
 @method_decorator(cache_no_user_data, name="dispatch")
 class ExploreView(TemplateView):
     template_name = "explore/explore.html"
+
+
+class AppView(View):
+    @xframe_options_exempt
+    @add_security_headers
+    def options(self, request, *args, **kwargs):
+        """
+        Handles OPTIONS requests which may be sent as "preflight CORS" requests
+        to check permissions.
+        """
+        return HttpResponse()
+
+    @cache_forever
+    @xframe_options_exempt
+    @add_security_headers
+    def get(self, request, app, path):
```

Comment:
It seems the app parameter can come with the `.zip` suffix already. In that case it can be renamed to `filename` or `zip_filename`, and the filename with the base path below could be `fullname`. Just a suggestion.

Reply:
But after reading the background TODO comment, maybe the parameter should be `app_name` and the filename could be:
```
        filename = os.path.join(base, app_name, "custom-channel-ui.zip")
```

Reply:
I was thinking about add the background inside the `zip` file and manage those background images in the custom web app repository.

In any case, if we place the background images on this repository we can add another api view to get that file, I think it's better to do not merge the zip view with other files.